search-service/src/main/kotlin/com/egm/stellio/search/service/EntityPayloadService.kt

```diff
@@ -103,6 +103,36 @@ class EntityPayloadService(
             .bind("specific_access_policy", specificAccessPolicy?.toString())
             .execute()
 
+    @Transactional
+    suspend fun mergeEntity(
+        entityId: URI,
+        ngsiLdAttributes: List<NgsiLdAttribute>,
+        expandedAttributes: ExpandedAttributes,
+        observedAt: ZonedDateTime?,
+        sub: Sub?
+    ): Either<APIException, UpdateResult> = either {
+        val createdAt = ZonedDateTime.now(ZoneOffset.UTC)
+        logger.debug("Merging entity {}", entityId)
+
+        val temporalEntityAttributes = temporalEntityAttributeService.getForEntity(entityId, emptySet())
```

Comment:
seems like an expensive operation to get data you'll often don't need. and in any case, it is better to do it in TemporalEntityAttributeService since it is part of the logic of merging entity attributes (EntityPayloadService does not care about this)